cmd/xl-v1-list-objects-heal.go

```diff
@@ -137,12 +137,18 @@ func (xl xlObjects) listObjectsHeal(bucket, prefix, marker, delimiter string, ma
 			result.Prefixes = append(result.Prefixes, objInfo.Name)
 			continue
 		}
-		result.Objects = append(result.Objects, ObjectInfo{
-			Name:    objInfo.Name,
-			ModTime: objInfo.ModTime,
-			Size:    objInfo.Size,
-			IsDir:   false,
-		})
+		// Check if the current object needs healing
+		nsMutex.RLock(bucket, objInfo.Name)
+		partsMetadata, errs := readAllXLMetadata(xl.storageDisks, bucket, objInfo.Name)
```

Comment:
Here I avoided to test on readAllXLMetadata() returned errors and I let xlShouldHeal() decides instead. I think that here we should show that there is something bad occurring and needs to be healed *even* when we are not able to heal it. @krishnasrinivas, what do you think?